02_bubble/starfallz/main.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprint(out, bubble([]int{3, 2, 1, 5}))
+}
+
+func bubble(s []int) []int {
+	for i := 0; i < len(s); i++ {
+		isSwapped := false
+		for j := 0; j < len(s)-1; j++ {
+			swap(s, j, &isSwapped)
+		}
+
+		if !isSwapped {
+			break
+		}
+	}
+
+	return s
+}
+
+func swap(s []int, j int, isSwapped *bool) {
```

Comment:
the function name is confusing.
This function could be named `swapIfNextValIsGrater`, but this has no sense in my mind.
1) I would suggest moving this logic into to `bubble` function
2) get rid of `tempVal` and implement swapping by sth like `s[i],s[i+1] = s[i+1],s[i]`

Reply:
Cool thanks for the shortcut for the tempVal thing.

I generally love separating concerns into smaller functions so I can unit test them in isolation. Although I do agree that the name could be better so I will rename the func if Ur cool with that.